crates/red_knot_python_semantic/src/types/mro.rs

```diff
@@ -328,6 +328,14 @@ impl<'db> ClassBase<'db> {
         Display { base: self, db }
     }
 
+    pub fn is_subtype_of(self, db: &'db dyn Db, target: ClassBase<'db>) -> bool {
+        match (self, target) {
+            (ClassBase::Any | ClassBase::Todo | ClassBase::Unknown, _) => false,
+            (_, ClassBase::Any | ClassBase::Todo | ClassBase::Unknown) => false,
+            (ClassBase::Class(class), ClassBase::Class(target)) => class.is_subclass_of(db, target),
+        }
+    }
```

Comment:
it looks like this is only used in one place. I'm not sure I love having this method -- a `ClassBase` is not the same concept as a `Type`, but having this method feels like it blurs the lines between the two enums in a slightly unfortunate way. Can we just inline it?

```diff
diff --git a/crates/red_knot_python_semantic/src/types.rs b/crates/red_knot_python_semantic/src/types.rs
index defa2dd9a..d49a67df3 100644
--- a/crates/red_knot_python_semantic/src/types.rs
+++ b/crates/red_knot_python_semantic/src/types.rs
@@ -666,9 +666,10 @@ impl<'db> Type<'db> {
             {
                 self_class.class.is_subclass_of_base(db, target_class.base)
             }
-            (Type::SubclassOf(self_class), Type::SubclassOf(target_class)) => {
-                self_class.base.is_subtype_of(db, target_class.base)
-            }
+            (
+                Type::SubclassOf(SubclassOfType { base: self_base }),
+                Type::SubclassOf(SubclassOfType { base: target_base }),
+            ) => Type::from(self_base).is_subtype_of(db, Type::from(target_base)),
             (
                 Type::SubclassOf(SubclassOfType {
                     base: ClassBase::Class(self_class),
diff --git a/crates/red_knot_python_semantic/src/types/mro.rs b/crates/red_knot_python_semantic/src/types/mro.rs
index 75da7f7f3..131b512ed 100644
--- a/crates/red_knot_python_semantic/src/types/mro.rs
+++ b/crates/red_knot_python_semantic/src/types/mro.rs
@@ -328,14 +328,6 @@ impl<'db> ClassBase<'db> {
         Display { base: self, db }
     }
 
-    pub fn is_subtype_of(self, db: &'db dyn Db, target: ClassBase<'db>) -> bool {
-        match (self, target) {
-            (ClassBase::Any | ClassBase::Todo | ClassBase::Unknown, _) => false,
-            (_, ClassBase::Any | ClassBase::Todo | ClassBase::Unknown) => false,
-            (ClassBase::Class(class), ClassBase::Class(target)) => class.is_subclass_of(db, target),
-        }
-    }
-
     /// Return a `ClassBase` representing the class `builtins.object`
```